kyuubi-ctl/src/main/scala/org/apache/kyuubi/ctl/KyuubiCtlArguments.scala

```diff
@@ -96,24 +97,46 @@ class KyuubiCtlArguments(args: Seq[String], env: Map[String, String] = sys.env)
       fail("Only support expose Kyuubi server instance to another domain")
     }
     validateZkArguments()
+
+    val defaultNamespace = defaultKyuubiConf.getOption(HA_ZK_NAMESPACE.key)
+    if (defaultNamespace.isEmpty || defaultNamespace.get.equals(nameSpace)) {
+      fail(
+        s"""
+          |Only support expose Kyuubi server instance to another domain, but the default
+          |namespace is [$defaultNamespace] and specified namespace is [$nameSpace]
+        """.stripMargin)
+    }
   }
 
   private def validateGetDeleteActionArguments(): Unit = {
     validateZkArguments()
     validateHostAndPort()
     validateUser()
+    mergeArgsIntoKyuubiConf()
   }
 
   private def validateListActionArguments(): Unit = {
     validateZkArguments()
+    mergeArgsIntoKyuubiConf()
+
+  }
+
+  private def mergeArgsIntoKyuubiConf(): Unit = {
+    defaultKyuubiConf.set(HA_ZK_QUORUM.key, zkQuorum)
+    defaultKyuubiConf.set(HA_ZK_NAMESPACE.key, nameSpace)
   }
 
+
   private def validateZkArguments(): Unit = {
```

Comment:
I guess we don't need this check only for a `null` case these variables can still be invalid after this check

Reply:
I tried that using a kyuubi conf without zk quorum to create a zk client, it will failed but after a timeout.

![image](https://user-images.githubusercontent.com/6757692/114287832-dd1a5c00-9a9c-11eb-9f6e-88f9f5ac4cd1.png)

Reply:
And for this one, if we do not check namespace in advance, the namespace would be "/", and might also stuck when creating zk client.
So, I think the zk args validation could help to prevent some error.

![image](https://user-images.githubusercontent.com/6757692/114287957-f8399b80-9a9d-11eb-8038-ccaa7134f2c2.png)